### PR TITLE
Staging for version increment automation.

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
This is the staging PR for to execute workflows that auto raise the version increment PR's
Example: https://github.com/prudhvigodithi/common-utils/pull/1
 
### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/1375
From solution: https://github.com/opensearch-project/opensearch-build/issues/1375#issuecomment-1159476686
Related issue: https://github.com/opensearch-project/common-utils/issues/193
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
Is the change of `System.getProperty` for `project.property` really necessary as part of this?

File: build.gradle [hunk 1]
```diff
@@ -140,6 +140,15 @@ shadowJar {
     classifier = null
 }
 
+task setVersion() {
+    if(project.hasProperty('newVersion')) {
```
Comment (dblock) [inline build.gradle:144]: Is this `if` necessary? Task should fail if the property is missing, shouldn't it?

Comment (prudhvigodithi) [reply build.gradle:144]: Hey @dblock added `if` to make sure task wont fail if not passed `-PnewVersion=x.x.x-SNAPSHOT`, assuming this task will be called from the [workflow](https://github.com/opensearch-project/opensearch-build/pull/2291), which should have `-PnewVersion`.

Error removing if condition with passed in `-PnewVersion`
* What went wrong:
A problem occurred evaluating root project 'opensearch-job-scheduler'.
> Could not get unknown property 'newVersion' for task ':setVersion' of type org.gradle.api.DefaultTask.

I see both pros and cons removing the if condition, so open for the change.

Comment (dblock) [reply build.gradle:144]: I think you want to use `onlyIf`:

```
task setVersion {
    onlyIf { project.hasProperty('newVersion') }
    ant...
    doLast {
        println "Setting version to ${project.getProperty('newVersion')}."
    }
}
```

**Comment by prudhvigodithi** [comment]
> Ok, I think I understand. The real reason is that `ant.propertyfile` is easier to use than some way of replacing a version inside build.gradle. I think you should push the property change separately, either before, or after the version increment.

Yes @dblock without having property change  we can still proceed to version increment, We solely rely on `opensearch_version = System.getProperty("opensearch.version", "3.0.0-SNAPSHOT")` the default value (here its `3.0.0-SNAPSHOT`)  and replace with version that is out there in OpenSearch core repo
The downside with this is we will not have a fixed project property `opensearch.version` where we can compare the version against with.
Example version increment PR I have tested https://github.com/prudhvigodithi/sql/pull/8/files

**Comment by prudhvigodithi** [comment]
Hey @dblock Pushed the change in my latest commit, can you please check the task now considering the above inputs.
Thank you
Sample common-utils [AUTO] PR: https://github.com/prudhvigodithi/common-utils/pull/2/files
@bbarani

**Comment by prudhvigodithi** [comment]
Thanks @dblock for all the details, I see now why onlyIF is good use case here. I have pushed the change in my latest commit.
```
./gradlew versionIncrement
> Task :versionIncrement SKIPPED

./gradlew versionIncrement -DnewVersion=1.3.3-SNAPSHOT
> Task :versionIncrement
Setting version to 1.3.3-SNAPSHOT.

./gradlew build nor ./gradlew build -DnewVersion=1.3.4-SNAPSHOT
(does not run versionIncrement task)

```

File: build.gradle [hunk 2]
```diff
@@ -140,6 +140,20 @@ shadowJar {
     classifier = null
 }
 
+task versionIncrement {
+    onlyIf { System.getProperty('newVersion') }
+    doLast {
+        ext.newVersion = System.getProperty('newVersion')
+        println "Setting version to ${newVersion}."
+        final String workingDir = project.buildDir.toString() + "/../"
+        ant.replaceregexp(match: opensearch_version.tokenize('-')[0], replace: newVersion.tokenize('-')[0], flags:'g', byline:true) {
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply build.gradle:150]: Thanks, added some comments @dblock

**Review by dblock — APPROVED** [review]
Lgtm

**Review by praveensameneni — DISMISSED** [review]

**Comment by dblock** [comment]
I imagine you're renaming tasks following https://github.com/opensearch-project/opensearch-plugin-template-java/pull/32?

**Comment by prudhvigodithi** [comment]
Ye @dblock I will be shortly pushing the code renaming the task as `updateVersion`.

**Comment by prudhvigodithi** [comment]
Thanks @dblock I have raised a new PR to `main` branch with backport labels, please advise if I can close this?
https://github.com/opensearch-project/common-utils/pull/200
Thank you

**Comment by dblock** [comment]
Yep, use backports.

**Pull request closed without merging**